include/alpaka/core/UniformCudaHip.hpp

```diff
@@ -85,89 +81,53 @@ namespace alpaka::uniform_cuda_hip::detail
-                    __FILE__,                                                                                         \
-                    __LINE__);                                                                                        \
-                ::alpaka::uniform_cuda_hip::detail::rtCheckIgnore<TApi, true>(                                        \
-                    cmd,                                                                                              \
-                    #cmd,                                                                                             \
-                    __FILE__,                                                                                         \
-                    __LINE__,                                                                                         \
-                    ##__VA_ARGS__);                                                                                   \
-            } while(0)
-#        if BOOST_COMP_CLANG
-#            pragma clang diagnostic pop
-#        endif
+#    define ALPAKA_UNIFORM_CUDA_HIP_RT_CHECK_IGNORE(cmd, ...)                                                         \
+        do                                                                                                            \
+        {                                                                                                             \
+            ::alpaka::uniform_cuda_hip::detail::rtCheckLastError<TApi, true>(                                         \
+                "'" #cmd "' A previous API call (not this one) set the error ",                                       \
+                __FILE__,                                                                                             \
+                __LINE__);                                                                                            \
+            ::alpaka::uniform_cuda_hip::detail::rtCheckIgnore<TApi, true>(                                            \
+                cmd,                                                                                                  \
+                #cmd,                                                                                                 \
+                __FILE__,                                                                                             \
+                __LINE__,                                                                                             \
+                {__VA_ARGS__});                                                                                       \
+        } while(0)
+#    if BOOST_COMP_CLANG
+#        pragma clang diagnostic pop
 #    endif
 
 //! CUDA/HIP runtime error checking with log and exception.
 #    define ALPAKA_UNIFORM_CUDA_HIP_RT_CHECK(cmd) ALPAKA_UNIFORM_CUDA_HIP_RT_CHECK_IGNORE(cmd)
```

Comment:
handle clang diagnostics should not be required if we change this to 

```
#define ALPAKA_UNIFORM_CUDA_HIP_RT_CHECK(cmd) ALPAKA_UNIFORM_CUDA_HIP_RT_CHECK_IGNORE(cmd,)
# define ALPAKA_UNIFORM_CUDA_HIP_RT_CHECK_NOEXCEPT(cmd) 
 ALPAKA_UNIFORM_CUDA_HIP_RT_CHECK_IGNORE_NOEXCEPT(cmd, )
```

If the user is calling `ALPAKA_UNIFORM_CUDA_HIP_RT_CHECK_IGNORE()` or `ALPAKA_UNIFORM_CUDA_HIP_RT_CHECK_IGNORE_NOEXCEPT` without any ignore list the warning will be shown again but this is nothing alpaka needs to handle.

Reply:
I am fine with the current PR but would like to point out that we most likely do not need to handle compiler diagnostics here.

Reply:
I just retained the warning suppression since it was there before. But your suggestion with `...IGNORE(cmd,)` is gold! That got rid of the warning entirely!